editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -53,8 +53,16 @@
 #include "scene/main/viewport.h"
 #include "scene/resources/packed_scene.h"
 
-#define MIN_ZOOM 0.01
-#define MAX_ZOOM 100
+// Min and Max are power of two in order to play nicely with successive increment.
+// That way, we can naturally reach a 100% zoom from boundaries.
+#define MIN_ZOOM 1. / 128
+#define MAX_ZOOM 128
+
+// Base increment factor defined as the twelveth root of two.
+// This allow a smooth geometric evolution of the zoom, with the advantage of
+// visiting all integer power of two scale factors.
+// note: this is analogous to the 'semitones' interval in the music world
+#define ZOOM_INCREMENT 0.0594630943592953 // 2**(1/12) - 1
```

Comment:
Alternative: I can replace the numerical value by `(Math::log(2, 1.0 / 12) - 1)`, which is more readable

Reply:
We now allow `constexpr` to be used in the C++ code, but I don't think `Math::log()` can be used in a `constexpr` context, or can it?

Reply:
Well, Math::log is not marked constexpr, but all my tests in compiler explorer shows that the compiler do eval those static values...
I personally prefer the Math::log, but it's up to you, as I'm well too young in this project, not yet familiar with styles and such...
Also bis, where can I find the information about accepted standards per version ? I have only seen the c++03 somewhere, so I didn't even proposed the constexpr version ^^

Reply:
> Also bis, where can I find the information about accepted standards per version ? I have only seen the c++03 somewhere, so I didn't even proposed the constexpr version ^^

We accept a subset of C++17 in the `master` branch, and full C++03 in stable branches like `3.2`. The subset in question is not formally written anywhere yet, but it boils down to "everything except `auto`, lambdas and `for` range loops" (more or less).